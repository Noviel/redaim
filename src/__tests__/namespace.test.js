import createTarget from '../target';
import createNamespace, { createNamespaceObject } from '../namespace';

describe('createNamespace', () => {
  const targetReducer = createTarget({
    'ADD': (state, action) => state + action.payload,
    'SUB': (state, action) => state - action.payload
  });

  const action = (target, exec, payload, type = 'TRIGGER') => ({
    type,
    meta: {
      exec,
      target
    },
    payload
  });

  it('should return a function', () => {
    const result = createNamespace({}, 'TRIGGER');

    expect(typeof result).toMatchSnapshot();
  });

  it('should throw an error with not a string trigger', () => {
    expect(() => {
      createNamespace({}, undefined);
    }).toThrow();
  });

  it('should not modify state with not corresponding action type', () => {
    const ns = createNamespace({
      targetOne: targetReducer(5),
      targetTwo: targetReducer(-100)
    }, 'TRIGGER');

    let state = ns(undefined, action('targetOne', 'ADD', 5, 'SOME_OTHER_ACTION_TYPE'));
    state = ns(state, action('targetTwo', 'SUB', 5, 'SOME_OTHER_ACTION_TYPE'));

    const toMatch = [state.targetOne, state.targetTwo];

    expect(toMatch).toMatchSnapshot();
  });

  it('should not modify state if target is not in namespace', () => {
    const ns = createNamespace({
      targetOne: targetReducer(5),
      targetTwo: targetReducer(-100)
    }, 'TRIGGER');

    const state = ns(undefined, action('targetUnknown', 'ADD', 5));

    const toMatch = [state.targetOne, state.targetTwo];

    expect(toMatch).toMatchSnapshot();
  });

  it('should apply actions to correct target', () => {
    const ns = createNamespace({
      targetOne: targetReducer(5),
      targetTwo: targetReducer(-100)
    }, 'TRIGGER');

    let state = ns(undefined, action('targetOne', 'ADD', 5));
    state = ns(state, action('targetTwo', 'SUB', 5));

    const toMatch = [state.targetOne, state.targetTwo];

    expect(toMatch).toMatchSnapshot();
  });  
});

describe('createNamespaceObject', () => {
  it('should return object with namespace as a key', () => {
    const namespaceObj = createNamespaceObject({}, 'TRIGGER');

    expect(Object.keys(namespaceObj)).toMatchSnapshot();
  });
});
