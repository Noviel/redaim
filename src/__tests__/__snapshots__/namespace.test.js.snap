// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNamespace should apply actions to correct target 1`] = `
Array [
  10,
  -105,
]
`;

exports[`createNamespace should not modify state if target is not in namespace 1`] = `
Array [
  5,
  -100,
]
`;

exports[`createNamespace should not modify state with not corresponding action type 1`] = `
Array [
  5,
  -100,
]
`;

exports[`createNamespace should return a function 1`] = `"function"`;

exports[`createNamespaceObject should return object with namespace as a key 1`] = `
Array [
  "TRIGGER",
]
`;
